thalo-eventstoredb/src/event_store.rs

```diff
@@ -0,0 +1,340 @@
+
+            // TODO: - can we try event eventlope ids as uuid in addition to usize?
+            // let uuid = event_data.id.clone();
+            let sequence = event_data.revision as usize;
+            if ids.contains(&sequence) {
+                let event_payload = event_data
+                    .as_json::<ESDBEventPayload>()
+                    .map_err(Error::DeserializeEvent)?
+                    .event_envelope::<A>(sequence)?;
+
+                rv.push(event_payload);
+            }
+        }
+
+        Ok(rv)
+    }
+
+    async fn load_aggregate_sequence<A>(
+        &self,
+        id: &<A as Aggregate>::ID,
+    ) -> Result<Option<usize>, Self::Error>
+    where
+        A: Aggregate,
+    {
+        let options = ReadStreamOptions::default()
+            .position(StreamPosition::End)
+            .max_count(1);
+
+        let events = self.read_stream(self.stream_id::<A>(Some(id)), options).await?;
+        if let Some(event) = events.get(0) {
+            let event_data = event.get_original_event();
+            return Ok(Some(event_data.revision as usize));
+        }
+
+        Ok(None)
+    }
+
+    async fn save_events<A>(
+        &self,
+        id: &<A as Aggregate>::ID,
+        events: &[<A as Aggregate>::Event],
+    ) -> Result<Vec<usize>, Self::Error>
+    where
+        A: Aggregate,
+        <A as Aggregate>::Event: serde::Serialize,
+    {
+        if events.is_empty() {
+            return Ok(vec![]);
+        }
+
+        let sequence = self.load_aggregate_sequence::<A>(id).await.unwrap_or(None);
+
+        let revision = {
+            match sequence {
+                Some(n) => ExpectedRevision::Exact(n as u64),
+                None => ExpectedRevision::NoStream,
+            }
+        };
+
+        let mut payload: Vec<EventData> = vec![];
+
+        for event in events.iter() {
+            let event_data_payload = ESDBEventPayload {
+                created_at: Utc::now(),
+                aggregate_type: <A as TypeId>::type_id().to_string(),
+                aggregate_id: id.to_string(),
+                event_data: serde_json::to_value(event).map_err(Error::SerializeEvent)?,
+            };
+
+            let event_data = EventData::json(event.event_type(), &event_data_payload)
+                .map_err(Error::SerializeEventDataPayload)?
+                .id(Uuid::new_v4());
+
+            payload.push(event_data);
+        }
+
+        let options = AppendToStreamOptions::default().expected_revision(revision);
+        let res = self
+            .client
+            .append_to_stream(&self.stream_id::<A>(Some(id)), &options, payload)
+            .map_err(|err| Error::WriteStreamError(sequence.unwrap_or(0) as usize, err))
+            .await?;
+
+        let end_position = res.next_expected_version as usize;
+        let start_position = end_position - (events.len() - 1);
+
+        Ok((start_position..end_position + 1).collect::<Vec<usize>>())
+    }
+}
+
+impl Debug for ESDBEventStore {
```

Comment:
I don't think this `Debug` implementation adds much information. I think it could be removed and doesn't need to implement Debug.